src/rlookup.c

```diff
@@ -599,10 +599,24 @@ static RSValue *replyElemToValue(RedisModuleCallReply *rep, RLookupCoerceType ot
   }
 }
 
+static bool isValueAvailable(const RLookupKey *kk, const RLookupRow *dst, RLookupLoadOptions *options) {
+  if (!options->forceLoad) {
+    if (kk->flags & RLOOKUP_F_VAL_AVAILABLE) {
+      // No need to "write" this key. It's always implicitly loaded!
+      return true;
+    } else if ((kk->flags & (RLOOKUP_F_SVSRC | RLOOKUP_F_ISLOADED)) == (RLOOKUP_F_SVSRC | RLOOKUP_F_ISLOADED)) {
```

Comment:
Why do we need to check if the key `RLOOKUP_F_ISLOADED`?

1. `RLOOKUP_F_SVSRC` and `RLookup_GetItem(kk, dst) == NULL` is enough evidence we don't need to load the value right?
    - have we checked what is the normalized form of an empty string, a string with only whitespace? If it's NULL we might have a problem
3. We shouldn't be here if the key is marked with `RLOOKUP_F_ISLOADED`. It means that a loader from the upstream already loaded the value, so following loaders that load individual keys shouldn't try to load it again.
    - see `(key->flags & RLOOKUP_F_ISLOADED &&       !(flags &  RLOOKUP_F_OVERRIDE))` condition in `GetKey`